Enable auto-reloading in watch mode when translations change

carlobeltrame opened the pull request:
The loaders already call addDependency with the directory as the argument, but the Webpack documentation states that it should be a filename. So by passing each detected file to addDependency individually, we gain automatic reloading for the most common case of an existing translation file changing.

Fixes https://github.com/kirschbaum-development/laravel-translations-loader/issues/21
Fixes https://github.com/kirschbaum-development/laravel-translations-loader/issues/6

luisdalmolin commented:
@carlobeltrame Thanks for the PR. The code looks good to me, but the test suite is failing with the new dependency added. If you rebase with the latest master, it should run the tests in this PR once you push again. If we get the build passing, I'm happy to merge this PR!

carlobeltrame commented:
I fixed the tests. I couldn't find any library for mocking in the dependencies, so I used a trivial manual mock.

I also added a new test for the added functionality. Since you don't use a dedicated expectation library, and the GitHub actions are still configured to run with node.js version 8 (which is not supported since 2020, and doesn't even have `assert.regex()` or `Array.slice()`), I had to jump through a few hoops to make it work. Let me know if you'd like me to squash the commits.

luisdalmolin reviewed (approved):
Nice work!

luisdalmolin commented:
Published 1.2.0 (https://github.com/kirschbaum-development/laravel-translations-loader/releases/tag/1.2.0).

andreiio commented:
@luisdalmolin could you also publish it to npm? please and thank you!

luisdalmolin commented:
@andreiio Oh yeah 🤦  - Done.
